### PR TITLE
Add Tiny JS World task

### PR DESCRIPTION
**Pull request by AriadnaKravchuk**
### A Tiny JS World

[Demo](https://ariadnakravchuk.github.io/a-tiny-JS-world) |
[Code base](https://github.com/AriadnaKravchuk/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by AriadnaKravchuk** [comment]
I completed self-checks

File: submissions/AriadnaKravchuk/tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   friends: ["Dominic", "Selina"]
+};
+
+const catWoman = {
+   species: "human",
+   name: "Selina",
+   gender: "female",
+   legs: 2,
+   hands: 2,
+   saying: cat.saying,
```
Comment (artem-trubin) [inline submissions/AriadnaKravchuk/tiny-js-world/index.js:56]: If we change `cay.saying`, would `catWoman.saying` change as well? The task requires catwoman to always say what the cat does.

**Review by artem-trubin — CHANGES_REQUESTED** [review]
Hey, nice work! Let's add an improvement, check the comment below.

**Comment by artem-trubin** [comment]
It works now, good job!

**Comment by AriadnaKravchuk** [comment]
> It works now, good job!

Thanks, and I have a question. What constants should be written in uppercase? (or I need use uppercase for all constant).

**Comment by artem-trubin** [comment]
> > It works now, good job!
> 
> Thanks, and I have a question. What constants should be written in uppercase? (or I need use uppercase for all constants).

## ALL UPPERCASE. LIKE THIS.

In all uppercase are usually those constants that are like "settings" if you know what I mean. Those that you write globally in the start of the file, like "MAIN_BUTTON_COLOR". Something that will 100% never change in the program and if it should be changed, you would go to this constant and change it there by hand. Or, sometimes, if you want to have some kind of string-based state, like a status or something. Imagine a flight as an object and you want to add a property `flight.status` that can be one of some different string statuses, like `in-progress`, `landed`, `setting-off` or something like this. To prevent yourself from making a simple mistake in these strings further in code, these could be defined as global constants like this:
```js
const STATUS_IN_PROGRESS = 'in-progress'
const STATUS_LANDED = 'landed'
const STATUS_SETTING_OFF = 'setting-off`

if (flight.status === STATUS_IN_PROGRESS) { // ...
```

This gives you several advantages:
1. When you type `STATUS_`, your code editor probably hands you all statuses as a hint, since they're in the global scope.
2. Protects you from typos on the building stage, since you'll just get an error if you write a non-existant variable name.

## FirstLetterUppercase. LikeThis.
This style is usually used to describe a class or a constructor function. Something that can later spawn instances of itself. Like **I**nhrabitant can spawn **i**nhabitants.

I hope I was descriptive enough, feel free to ask more questions. Here's a cute possum for your curiosity:

image

**Comment by AriadnaKravchuk** [comment]
> > > It works now, good job!
> > 
> > 
> > Thanks, and I have a question. What constants should be written in uppercase? (or I need use uppercase for all constants).
> 
> ## ALL UPPERCASE. LIKE THIS.
> In all uppercase are usually those constants that are like "settings" if you know what I mean. Those that you write globally in the start of the file, like "MAIN_BUTTON_COLOR". Something that will 100% never change in the program and if it should be changed, you would go to this constant and change it there by hand. Or, sometimes, if you want to have some kind of string-based state, like a status or something. Imagine a flight as an object and you want to add a property `flight.status` that can be one of some different string statuses, like `in-progress`, `landed`, `setting-off` or something like this. To prevent yourself from making a simple mistake in these strings further in code, these could be defined as global constants like this:
> 
> ```js
> const STATUS_IN_PROGRESS = 'in-progress'
> const STATUS_LANDED = 'landed'
> const STATUS_SETTING_OFF = 'setting-off'
> 
> if (flight.status === STATUS_IN_PROGRESS) { // ...
> ```
> 
> In fact, you're still operating string values, but in code - they're wrapped in constants.
> 
> This gives you several advantages:
> 
> 1. When you type `STATUS_`, your code editor probably hands you all statuses as a hint, since they're in the global scope.
> 2. Protects you from typos on the building stage, since you'll just get an error if you write a non-existant variable name.
> 
> ## FirstLetterUppercase. LikeThis.
> This style is usually used to describe a class or a constructor function. Something that can later spawn instances of itself. Like **I**nhrabitant can spawn **i**nhabitants.
> 
> I hope I was descriptive enough, feel free to ask more questions. Here's a cute possum for your curiosity:
> 
> image

So, object like cat, dog etc should be written in FirstLetterUppercase?

**Comment by artem-trubin** [comment]
No, regular objects are lowercase. If it's a class or a constructor function (in other words, something that can create instances) - then we use FirstLetterUppercase.

**Comment by AriadnaKravchuk** [comment]
> No, regular objects are lowercase. If it's a class or a constructor function (in other words, something that can create instances) - then we use FirstLetterUppercase.

Okay, thank you!

**Pull request merged**